pulser/json/coders.py

```diff
@@ -12,20 +12,20 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from __future__ import annotations
 import importlib
 import inspect
 from json import JSONEncoder, JSONDecoder
+from typing import Dict, Any, Union
 
 import numpy as np
 
 from pulser.json.utils import obj_to_dict
 from pulser.parametrized import Variable
 
-from typing import Dict, Any
-
 
 class PulserEncoder(JSONEncoder):
-    def default(self, o: Any) -> Dict:
+    def default(self, o: Any) -> Union[Dict[str, Any], Any]:
```

Comment:
Doesn't just `Dict[str, Any]` work? A union with `Any` is a bit redundant, is the same as being `Any`

Reply:
Without `Any` it's showing `error: Returning Any from function declared to return "Dict[str, Any]"` in following lines of code - 
* https://github.com/pasqal-io/Pulser/blob/954fef70885a498df8d747c430b4c7d528e93615/pulser/json/coders.py#L28
* https://github.com/pasqal-io/Pulser/blob/954fef70885a498df8d747c430b4c7d528e93615/pulser/json/coders.py#L36

The return types are similar to a dictionary. I think casting will be a better option.

Reply:
Done